clients/hadoop-to-venice-bridge/src/main/java/com/linkedin/venice/hadoop/input/kafka/VeniceKafkaInputReducer.java

```diff
@@ -51,10 +51,9 @@ protected Optional<VeniceWriterMessage> extract(
     if (!valueIterator.hasNext()) {
       throw new VeniceException("There is no value corresponding to key bytes: " + ByteUtils.toHexString(keyBytes));
     }
-    List<KafkaInputMapperValue> mapperValues = getMapperValues(valueIterator);
     return isChunkingEnabled()
-        ? extractChunkedMessage(keyBytes, mapperValues)
-        : extractNonChunkedMessage(keyBytes, mapperValues);
+        ? extractChunkedMessage(keyBytes, getMapperValues(valueIterator))
```

Comment:
Would you like to leave a TODO message about how to optimize memory usage for chunked stores?

Reply:
Hoping to take a stab at it tomorrow, but wanted to keep the current change surgical.

Reply:
Marked `getMapperValues` function as deprecated and left a TODO.